Source/CesiumRuntime/Public/Cesium3DTileset.h

```diff
@@ -819,6 +820,8 @@ class CESIUMRUNTIME_API ACesium3DTileset : public AActor {
 private:
   TUniquePtr<Cesium3DTilesSelection::Tileset> _pTileset;
 
+  FMetadataDescription encodedMetadataDescription;
```

Comment:
@kring I think that's how I implemented it initially but I don't think it's safe to access components in the load thread, so at the minimum the metadata description needs to be copied over to the tileset on LoadTileset. Accessing a description housed on the tileset from the load thread is thread-safe because the tileset destructor waits for all async tasks to finish before destruction. So all references to the metadata description, from the gltf and primitive components will be gone before the tileset along with the description are deleted.

Maybe there could be a better API to retrieve the description from the tileset on any thread, so that the resource preparer does not need to be friends. Would that be better?

Reply:
Anytime we find ourselves making case a like "this is thread-safe because..." we should stop and consider whether we're approaching this correctly. Because:

1. Proving thread-safety is hard. It's easy to miss something.
2. Seemingly innocent changes later can invalidate the argument that enabled us to convince ourselves it was thread-safe. We might not realize this invalidation has happened and end up creating an intermittent and hard to debug failure, maybe without even realizing it until months later.

For example, it's not enough to be sure the lifetime of the object you're accessing from the worker thread will be long enough. We also have to be sure that that object won't change at all. Because any change in one thread while accessing from another thread can (unless we're extremely careful) lead to undefined behavior.

I agree with you in this case that it's safe. Still, why rely on that unless we have to? In this case the problem isn't that copying for the worker thread is expensive (it's not); the problem is how to copy it to the worker thread at all? I think the answer to that is to use the same pattern we just adopted for raster overlays, where `prepareRasterInLoadThread` takes a `rendererOptions` parameter that is set on RasterOverlayOptions and then passed by copy to the worker thread. We should do the same thing with `prepareInLoadThread` and `TilesetOptions`.  Then `alwaysIncludeTangents` can go in the renderer options as well, and we completely eliminate the need to access `this` from the worker.

Reply:
I wrote https://github.com/CesiumGS/cesium-native/issues/480, it doesn't have to be part of this PR.